modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/marketplace-base-fragments/fragments/app-banner/index.js

```diff
@@ -37,3 +63,50 @@ const getSiteURL = () => {
 
 	return '';
 };
+
+const getProducts = async () => {
+	const products = fetcher(
+		`/o/headless-commerce-admin-catalog/v1.0/products`
+	);
+
+	return products;
+};
+
+const getUrlString = (url) => {
+	const urlSplit = url.split('/');
+	const filteredUrl = urlSplit[urlSplit.length - 1];
+
+	return filteredUrl;
+};
+
+const setLabels = async () => {
+	const currentUrl = Liferay.currentURL;
+	const labelDiv = document.getElementById('categoryLabel');
+
+	const products = (await getProducts()).items;
+
+	const filteredLabels = products.reduce((labels, product) => {
+		if (product.urls.en_US === getUrlString(currentUrl)) {
+			const categoryTags = product.categories
+				.filter(
+					(category) =>
+						category.vocabulary === 'marketplace solution category'
+				)
+				.map(
+					(category) =>
+						`<span class="banner__product-tag rounded mb-2 mr-2 py-2 px-3">${category.name}</span>`
+				)
+				.join('');
+
+			if (categoryTags) {
+				labels.push(categoryTags);
+			}
+		}
+
+		return labels;
+	}, []);
+
+	labelDiv.innerHTML = filteredLabels;
+};
+
+setLabels();
```

Comment:
If the article above makes sense, you can change it here too:

```suggestion
setCategoryLabels();
```